features/0804-didcomm-rpc/README.md

```diff
@@ -0,0 +1,381 @@
+
+- request-received
+- abandoned
+- completed
+
+The state transition table for the `responder` is:
+
+| State / Events   | Receive Request                        | Send Response                  | Send Ack                       | Send Problem Report             |
+| ---------------- | -------------------------------------- | ------------------------------ | ------------------------------ | ------------------------------- |
+| *Start*          | Transition to <br>**request-received** |                                |                                |                                 |
+| request-received |                                        | Transition to <br>**complete** | Transition to <br>**complete** | Transition to <br>**abandoned** |
+| completed        |                                        |                                |                                |                                 |
+| abandoned        |                                        |                                |                                |                                 |
+
+### Messages
+
+The following are the messages in the DRPC protocol, including the adopted messages.
+
+#### Request Message
+
+The `request` message is sent by the `sender` to initiate the protocol. The
+message contains the information necessary for the `responder` to process the
+request, prepare the response, and send the response message back. It is assumed
+the `sender` knows what types of requests the `responder` is prepared to receive
+and process. How the `sender` and `responder` coordinate that understanding is
+out of scope of this protocol.
+
+The `request` message items have been chosen to make putting an HTTP request
+into the `request` message easy (borrowing from [RFC 0335 HTTP Over
+DIDComm]), but an HTTP request is not required. As long as `request_type`
+identifies the request being made, and the `responder` understands it, the
+message can be processed.
+
+[RFC 0335 HTTP Over DIDComm]: /features/0335-http-over-didcomm/README.md
+
+```jsonc
+{
+  "@type": "https://didcomm.org/drpc/1.0/request",
+  "@id": "2a0ec6db-471d-42ed-84ee-f9544db9da4b",
+  "request": "<request>",
+  "headers": [{"name": "Content-Type", "value": "application/json"}],
+  "body.base64": "<Base64 Encoded JSON>"
+}
+```
+
+The items in the message are as follows:
+
+- `@type` -- required, must be as above
+- `@id` -- required, must be as defined in [RFC 0005 DIDComm]
+- `request_type` -- **required**, an identifier indicating the type of the request. The value could be a URI, such as an HTTP URL, that the `sender` would like `responder` to resolve, or the value could be just a string understood by the `responder`.
+- `headers` -- optional, a JSON array typically used for conveying HTTP headers to be used in the processing of an HTTP URL `request_type`. As with all parts of this protocol, the meaning is up to the participants of an instance of the protocol.
```

Comment:
I think RPC has no need to be related to HTTP style headers. Besides content type, do you have specific needs for this? We can solve that part with attachments, and we could then just make this part of the body. So if you need headers, we make it part of the body?